### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/anope/commit/b888ff590d29bd13e7ce6eb0aee6bc40c39809bb#commitcomment-52291843

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/anope/commit/27bfed532a732bc2c457cfa7540d7c7804145776

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: include/account.h [hunk 1]
```diff
@@ -262,17 +262,17 @@ class CoreExport IdentifyRequest
 	 * for the request to complete. Multiple modules may hold a request at any time,
 	 * but the request is not complete until every module has released it. If you do not
 	 * require holding this (eg, your password check is done in this thread and immediately)
-	 * then you don't need to hold the request before Successing it.
+	 * then you don't need to hold the request before Succeeding it.
```
Comment (jsoref) [inline include/account.h:265]: This is probably wrong, but, the original is messy, possibly:
```suggestion
	 * then you don't need to hold the request before `Success`ing it.
```

File: include/account.h [hunk 2]
```diff
@@ -262,17 +262,17 @@ class CoreExport IdentifyRequest
...
 	void Hold(Module *m);
 
 	/** Releases a held request
-	 * @param m The module releaseing the hold
+	 * @param m The module releasing the hold
 	 */
 	void Release(Module *m);
 
-	/** Called by modules when this IdentifyRequest has successeded successfully.
+	/** Called by modules when this IdentifyRequest has succeeded successfully.
```
Comment (jsoref) [inline include/account.h:275]: This isn't great

File: src/tools/geoipupdate.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@
 # by the irc2sql module on the first load.
 
 # Don't forget to rename this file or your changes
-# will be overwritte on the next 'make install'
+# will be overwrite on the next 'make install'
```
Comment (SadieCat) [inline src/tools/geoipupdate.sh:9]:
```suggestion
# will be overwritten on the next 'make install'
```

File: modules/CMakeLists.txt [hunk 4]
```diff
@@ -117,7 +117,7 @@ macro(build_subdir)
 
       # Append this source file's linker flags to the subdirectoy's linker flags, if there are any to append
       if(TEMP_DEPENDENCIES)
-        append_to_list(SUBDIR_EXTRA_DEPENDS ${TEMP_DEPDENCIES})
```
Comment (jsoref) [inline modules/CMakeLists.txt:120]: This appears to be a bug

File: include/account.h (hunk 2 above)
Comment (SadieCat) [inline include/account.h:275]:
```suggestion
	/** Called by modules when this IdentifyRequest has succeeded.
```

File: include/account.h (hunk 1 above)
Comment (SadieCat) [inline include/account.h:265]:
```suggestion
	 * then you don't need to hold the request before calling `Success()`.
```

File: include/account.h (hunk 2 above)
Comment (SadieCat) [inline include/account.h:275]:
```suggestion
	/** Called by modules when this IdentifyRequest has succeeded.
```

File: include/anope.h [hunk 5]
```diff
@@ -766,7 +766,7 @@ template<typename T> inline T convertTo(const Anope::string &s, bool failIfLefto
 }
 
 /** Casts to be used instead of dynamic_cast, this uses dynamic_cast
- * for debug builds and static_cast on releass builds
+ * for debug builds and static_cast on releases builds
```
Comment (SadieCat) [inline include/anope.h:769]:
```suggestion
 * for debug builds and static_cast on release builds
```

**Comment by SadieCat** [comment]
LGTM. Will merge when CI goes green.

Thanks for this, I'll definitely be looking into your action soon.

**Pull request merged**